server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java

```diff
@@ -101,23 +122,23 @@ public CatalogInfoDAO getCatalogDAO(Session session, String name) {
     public CatalogInfo updateCatalog(String name, UpdateCatalog updateCatalog) {
         ValidationUtils.validateSqlObjectName(updateCatalog.getNewName());
         // cna make this just update once we have an identifier that is not the name
-        try (Session session = sessionFactory.openSession()) {
+        try (Session session = SESSION_FACTORY.openSession()) {
             Transaction tx = session.beginTransaction();
             try {
-                CatalogInfoDAO catalogInfo = getCatalogDAO(session, name);
-                if (catalogInfo == null) {
+                CatalogInfoDAO catalogInfoDAO = getCatalogDAO(session, name);
+                if (catalogInfoDAO == null) {
                     throw new BaseException(ErrorCode.NOT_FOUND, "Catalog not found: " + name);
                 }
                 if (getCatalogDAO(session, updateCatalog.getNewName()) != null) {
                     throw new BaseException(ErrorCode.ALREADY_EXISTS,
                             "Catalog already exists: " + updateCatalog.getNewName());
                 }
-                catalogInfo.setName(updateCatalog.getNewName());
-                catalogInfo.setComment(updateCatalog.getComment());
-                catalogInfo.setUpdatedAt(new Date());
-                session.merge(catalogInfo);
+                catalogInfoDAO.setName(updateCatalog.getNewName());
+                catalogInfoDAO.setComment(updateCatalog.getComment());
+                catalogInfoDAO.setUpdatedAt(new Date());
+                session.merge(catalogInfoDAO);
```

Comment:
Updates are not implemented yet. Left 2 items for follow up PR since this one was already big: https://github.com/unitycatalog/unitycatalog/pull/89#issuecomment-2194175724